core/src/main/java/org/elasticsearch/common/network/NetworkAddress.java

```diff
@@ -180,4 +182,13 @@ static String format(InetAddress address, int port, boolean includeHost) {
         
         return builder.toString();
     }
+    
+    @SuppressForbidden(reason = "we call toString to avoid a DNS lookup")
+    private static String getInet6HostAddress(InetAddress address) {
+        return address.getHostAddress().replaceAll(":0+([0-9])",":$1")
+                .replaceAll("((?::0\\b){2,}):?(?!\\S*\\b\\1:0\\b)(\\S*)", "::$2")
+                .replaceFirst("^0::", "::")
+                .replaceFirst("%\\d+", "");
+        // see http://stackoverflow.com/a/7044170/4746692
```

Comment:
I don't think its good to implement address formatting with some regex from stackoverflow.

we need solid code that follows the RFCs: which is what the guava code does. We should not have to get "incorrect" formatting of internet addresses just to remove guava. We will keep guava around as long as we can until things are correct.